Automattic/wp-calypso
# Jetpack Cloud: Add site selector to new sidebar header

**NOTE: Because we're early in the development process, evaluate this PR as a proof-of-concept. It is almost certain the specifics and even the architecture will change before the new navigation sidebar is shipped to production.**

This PR adds a new `Header` component to the Jetpack navigation that's currently under development, which contains a link to the site selector and updated styling.

Resolves https://github.com/Automattic/jetpack-genesis/issues/38.

## Proposed Changes

* Create a new `Header` component to house the current site/all sites indicator (and eventually, the profile dropdown).
* Add new styling to the sidebar to properly display the header according to designs *(see: Ra0o7IPph3bCS2xUCnAusZ-fi-4361%3A51432)*.

## Implementation notes

Since I started work on this PR, @Automattic/jetpack-genesis and @sejas have since discussed the architecture of the new navigation sidebar. Discussions are ongoing, but the net effect is that the code in this PR may eventually need to be reshaped or moved to accommodate usage in Calypso Blue. The initial version you see here is written only with Jetpack Cloud in mind.

Also note that in this stage, navigation functionality is still in flux, so you may notice some slight irregularities when visiting different pages with this new sidebar. The more important criteria to evaluate here are the appearance of the sidebar and the 

## Testing Instructions

* In your Jetpack Cloud test environment, visit `/dashboard` and/or `/partner-portal/licenses`.
* Make sure to add `?flags=jetpack/new-navigation` to your browser's query string to enable the new sidebar.
* Verify that you see the `All Sites` indicator at the top of the sidebar, and that it visually matches designs (Ra0o7IPph3bCS2xUCnAusZ-fi-4361%3A51432).
* Verify that when you click on `All Sites`, the Site Selector is displayed and functions more-or-less appropriately (see implementation notes above).
* Select a site using the Site Selector. Verify you're shown the single-site navigation sidebar.
* Click on your site's name in the sidebar, then click All My Sites again. Verify you're returned to the All Sites view.

https://github.com/Automattic/wp-calypso/assets/670067/47ad6c7e-7c1a-49c9-a902-7b1f4fbf322d
